core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ChunkBuffer.java

```diff
@@ -88,6 +93,33 @@ public synchronized Iterable<ByteBuffer> split(ByteBuffer inputByteBuffer) {
         return byteBuffers;
     }
 
+    private synchronized Iterable<ByteBuffer> splitWithUnknownLength(ByteBuffer inputByteBuffer) {
+        boolean isLastChunk = inputByteBuffer.remaining() != chunkSize;
```

Comment:
What if this method gets invoked again? Then it would not be the last chunk, right?

Reply:
removed this method to use a single split() method